### ISSUE TITLE
feat(ngbot): missing `merge` label should count as "pending", not "failing"

### ISSUE THREAD

**Issue by gkalpak** [angular/angular#23644]
## I'm submitting a...
```
[x] Other... Please describe:
```

## Current behavior
While a PR doesn't have one of the required labels for merging (e.g. `PR action: merge`), it's `merge-status` appears as "failing". This affects the PR status shown at the bottom of the page and also the icon used by GitHub on the tab (which is useful for getting an overview of a PR's status without having to switch to the tab and scroll to the bottom). 

Showing the status as "failing" in this case is slightly confusing/misleading, because normally a failure means the PR author has to fix something. There is nothing to fix though, while waiting for a review or being blocked on another issue.
In other words, not having the `merge` label is not an error (99% of the time), but rather intentionally postponed until the PR is in an appropriate state. Thus, counting as "pending" is more accurate than "failing" imo.

## Expected behavior
Not having a required label for mergin should count as "pending" instead of "failing" as far as `merge-status` is concerned.

**Comment by gkalpak** [comment]
cc @IgorMinar, @manughub, @ocombe

**Comment by ocombe** [comment]
I don't know, for me "pending" means that we are still waiting on something and that we should come back later, while "failure" means that I should check it out and see what's the issue.
In this case, failure will make me look at the error message and add the required label, while pending will not.

**Comment by gkalpak** [comment]
Yes, but most of the time, you should not add the required label, because the PR has not been reviewed yet (so it is actually correct that it doesn't have the `merge` label).

How about this:
When any other status is red or pending (with the exception of `google3`), then missing the `merge` label counts as "pending" (or maybe not considered at all).
When everything else is green (and `google3` is green or pending), then missing the `merge` label counts as "failing".
